src/tsickle.ts

```diff
@@ -630,6 +638,21 @@ class Annotator extends ClosureRewriter {
     return toArray(reexports.keys());
   }
 
+  private writeModuleSpecifier(moduleSpecifier: ts.Expression) {
+    let moduleId = (moduleSpecifier as any).text;
+    if (this.tsOpts && this.host) {
+      const resolved = ts.resolveModuleName(moduleId, this.file.fileName, this.tsOpts, this.host);
+      if (resolved && resolved.resolvedModule) {
+        const resolvedModule = resolved.resolvedModule.resolvedFileName.replace(/(\.d)?\.ts$/, '');
+
+        if (resolvedModule.match(/\/index$/) && !moduleId.match(/\/index$/)) {
+          moduleId += '/index';
```

Comment:
Does this do the right thing if the person writes
`import * from 'foobar.js'`
?  I can't remember if they are trying to get people to do that now... 

...and do we even care?  Might be fine to just not worry about it for now.